src/components/StoreInfo/StoreLocation.tsx

```diff
@@ -17,15 +17,56 @@ const StoreLocation: FC<Props> = ({ locations }) => {
                   className={styles.wrap__singleLocation}
                   key={location.seller_id}
                 >
-                  <address>
-                    <p className={styles.address}>
-                      {location.address1 + (location.address2 || '')}
+                  <address
+                    itemProp="address"
+                    itemType="https://schema.org/PostalAddress"
+                  >
+                    <p itemProp="streetAddress" className={styles.address}>
+                      {location.address1}
+                      {location.address2 ? <br /> : ''}
+                      {location.address2 ? location.address2 : ''}
                     </p>
                     <p className={styles.address}>
-                      {location.city}, {location.state} {location.zip_code}
+                      <span itemProp="addressLocality">{location.city}</span>,{' '}
+                      <span itemProp="addressRegion">{location.state}</span>{' '}
+                      <span itemProp="postalCode">{location.zip_code}</span>
                     </p>
                   </address>
-                  <p className={styles.address}>{location.phone_number}</p>
+                  {location.address1 && (
+                    <span>
+                      <a
+                        href={
+                          'https://www.google.com/maps/place/' +
+                          (
+                            (location.address1 + ' ' || '') +
+                            (location.city + ' ' || '') +
+                            (location.state + ' ' || '') +
+                            (location.zip_code || '')
+                          ).replaceAll(' ', '+')
+                        }
+                        target="_blank"
+                        className={styles.mapURL}
+                      >
+                        <i className="fa fa-map-marker"></i> Map
+                      </a>
+                    </span>
+                  )}
+                  {location.phone_number && (
+                    <p itemProp="telephone" className={styles.phone}>
+                      {/^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im.test(
```

Comment:
I feel like we should move this regex somewhere less embedded -- at least maybe to the top of the component!

Reply:
good to add to the utils folder :)